client/src/main/scala/org/http4s/client/oauth1.scala

```diff
@@ -0,0 +1,102 @@
+package org.http4s
+package client
+
+import java.nio.charset.StandardCharsets
+
+import org.http4s.util.{UrlFormCodec, UrlCodingUtils}
+import org.http4s.headers.Authorization
+import org.http4s.util.string._
+
+import javax.crypto
+
+import scala.collection.mutable.ListBuffer
+import scalaz.concurrent.Task
+
+/** Basic OAuth1 message signing support
+  * 
+  * This feature is not considered stable.
+  */
+object oauth1 {
+
+  private val SHA1 = "HmacSHA1"
+  private def UTF_8 = StandardCharsets.UTF_8
+
+  val OutOfBounds = "oob"
+
+  case class Consumer(key: String, secret: String)
+  case class Token(value: String, secret: String)
+
+  def signRequest(req: Request, consumer: Consumer, callback: Option[Uri],
+                  verifier: Option[String], token: Option[Token]): Task[Request] = {
+    getUserParams(req).map { case (req, params) =>
+      val auth = genAuthHeader(req.method, req.uri, params, consumer, callback, verifier, token)
+      req.withHeaders(auth)
+    }
+  }
+
+  def genAuthHeader(method: Method, uri: Uri, userParams: Seq[(String,String)], consumer: Consumer,
```

Comment:
I would like to expose it, but perhaps mark it as unstable. Perhaps I've been listening to Edward Kmett to much, but it seems like a reasonable idea to expose it with a stern warning that its not considered stable.